src/rpc/rawtransaction.cpp

```diff
@@ -1035,7 +1036,7 @@ static UniValue sendrawtransaction(const JSONRPCRequest& request)
                 "\nAlso see createrawtransaction and signrawtransactionwithkey calls.\n",
                 {
                     {"hexstring", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The hex string of the raw transaction"},
-                    {"allowhighfees", RPCArg::Type::BOOL, /* default */ "false", "Allow high fees"},
+                    {"maxfeerate", RPCArg::Type::AMOUNT, /* default */ FormatMoney(maxTxFee), "Allow high fees (boolean) or reject transactions whose fee rate is higher than the specified value (numeric), expressed in " + CURRENCY_UNIT + "/kB\n"},
```

Comment:
If bool is no longer mentioned here, the transition should be mentioned in the release notes, though

Reply:
I'm hesitant to allow some hidden underlying default max fee when using a boolean. If we are removing the option, we should remove it completely, not just from the documentation.

I do realize that this means someone who use '1' as boolean true will accept any fees up to 1 BTC/kB, which is rather high. Maybe removing the boolean style completely is actually the way to go.

Reply:
> Maybe removing the boolean style completely is actually the way to go.

I tend to agree with your idea. An option could be to have a `-deprecatedrpc=sendrawtransaction -deprecatedrpc=testmempoolaccept` to toggle back to the bool behavior.

Reply:
> An option could be to have a `-deprecatedrpc=sendrawtransaction`

That's one way, although I think it may be overkill. How about catching if a bool is passed, and then returning an error message telling the user to set 'maxreerate' to 0, which maintains exactly the same behaviour. Code:

```
    // TODO: temporary migration code for old clients. Remove in v0.21
    if (request.params[1].isBool()) {
        throw JSONRPCError(RPC_INVALID_PARAMETER, "Second argument must be numeric (maxfeerate) and no longer supports a boolean. To allow a transaction with high fees, set maxfeerate to 0.");
    } else if (request.params[1].isNum()) {
        [...]
```
Or similar.

Reply:
I think the throw behavior makes sense, unless we hear complaints. I'll put that in. Thanks for code example.